sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/CtlWorkload.java

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.benchmark.linkedin;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.benchmark.Configuration;
+import com.azure.cosmos.benchmark.ScheduledReporterFactory;
+import com.azure.cosmos.benchmark.linkedin.data.Key;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.ScheduledReporter;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.base.Preconditions;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * TODOs
+ *  - Initialize the metrics reporter
+ */
+public class CtlWorkload {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CtlWorkload.class);
+
+    private final Configuration _configuration;
+    private final CosmosAsyncClient _client;
+    private final CosmosAsyncClient _bulkLoadClient;
+    private final MetricRegistry _metricsRegistry;
+    private final ScheduledReporter _reporter;
+    private final ResourceManager _resourceManager;
+    private final Map<Key, ObjectNode> _testData;
+    private final DataLoader _dataLoader;
+    private final GetTestRunner _getTestRunner;
+
+    public CtlWorkload(final Configuration configuration) {
+        Preconditions.checkNotNull(configuration, "The Workload configuration defining the parameters can not be null");
+
+        _configuration = configuration;
+        _client = AsyncClientFactory.buildAsyncClient(configuration);
+        _bulkLoadClient = AsyncClientFactory.buildBulkLoadAsyncClient(configuration);
```

Comment:
The client configuration on our side, for the READ path, sets some values such as retry-on-throttling to false, maxPool size etc. We want to preserve the same client configuration in this code, else we are comparing apples-to-oranges.

As for the BulkLoadAsyncClient, it is used in bulk loading data and 429s are common. To handle that, some of the client configurations are different e.g. we want to retry and have a higher timeout values. The 2 clients are created to decouple the data-loading from the read testing.

I had seen this configuration and explored the code in AsyncDocumentClient. I chose not to use it because it would deviate in how the client was currently configured.

Having explained the long story, if the recommendation to all clients is to leverage enableTransportClientSharing, then I can enable it. Let me know.

Reply:
To add to the previous comment, after the DataLoader/loadData completes, the BulkLoadAsyncClient can be closed in LICtlWorkload (in the setup()) method. Would that address the concerns raised (which make sense to me)

Reply:
Based on our conversation earlier today, I have updated the implementation to close the bulkLoadAsyncClient (which is part of the test setup) after the data loading is completed.